engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/util/CollectionUtil.kt

```diff
@@ -16,7 +16,7 @@ inline fun <reified T> Collection<*>.singleOfTypeOrNull(predicate: (T) -> Boolea
  * Like [singleOrNull] but the single item must be of type [T].
  */
 inline fun <reified T> Collection<*>.singleOfType(predicate: (T) -> Boolean = { true }): T {
-    return singleOfTypeOrNull(predicate) as T
```

Comment:
Ok this is where it went "wrong", kinda. I thought casting to a not null type would also do a null check, but no. It only does a plain Java cast. Then the error went undetected.